src/web/app/components/sortable-table/sortable-table.component.ts

```diff
@@ -9,6 +9,16 @@ export interface ColumnData {
   header: string;
   headerToolTip?: string;
   sortBy?: SortBy; // optional if the column is not sortable
+  component?: Type<any>; // Must use in conjunction with CustomTableCellData in rows
+}
+
+/**
+ * Data provided for custom component used in table cell
+ * Must use in conjunction with component in ColumnData
+ */
+export interface CustomTableCellData {
```

Comment:
I think essentially the PR wants to distinguish the data presentation from the data itself in the sortable table. Therefore, you may not want the `rows` to by of type `any[]`. Instead, we can define a new interface

```
export interface TableRow {
    value: any, // value used for sorting with sortBy provided in ColumnData
    displayableValue?: any; // displayable value 
    componentInput?: Record<string, any>; // @Input values for customized component if specified in ColumnData
}
```

By doing this, the data presentation (use dynamic component or just plain-string displayable value) is separated from the data (`value`) itself.

I think the current type definition `any[][] | CustomTableCellData[][]` is a bit confusing as it will enforce anything since we have `any`.

Reply:
Yeah I agree. It was quite awkward writing `any[][] | CustomTableCellData[][]`. 

Another question I have is whether I should provide the component every time in the individual cell data, or to specify it once in the column itself.

```
export interface TableRow {
    value: any, // value used for sorting with sortBy provided in ColumnData
    displayableValue?: any; // displayable value 
    component: Type<any>; // <-- Do I put this here?
    componentInput?: Record<string, any>; // @Input values for customized component if specified in ColumnData
}
```

Feels like it would introduce a stronger relationship between the `component` and the `componentInput`, which would make more sense?

Reply:
Let's keep it simple. Personally I do not foresee any possibility that we will use different components for different cells that belong to the same column.

Reply:
Actually I wasn't worrying about that. 

I felt that the relationship between the column and the row might not be clear enough to the caller. As you can see, I had to put a lot of comments to highlight to the user that the row and the column must be aligned, and the user must remember which cells uses which component from which column. Putting them in the same object structure would get rid of all these ambiguity

Reply:
Sure. We can also got for this solution.